rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/importexport/Sw360ImportExportService.java

```diff
@@ -0,0 +1,253 @@
+            }
+        }
+
+        ByteArrayInputStream byteArrayInputStream = CSVExport.createCSV(ReleaseLinkCSVRecord.getCSVHeaderIterable(), csvRows);
+        String filename = String.format(fileConstant, SW360Utils.getCreatedOn());
+        response.setHeader("Content-Disposition", String.format("Release; filename=\"%s\"", filename));
+        FileCopyUtils.copy(byteArrayInputStream, response.getOutputStream());
+    }
+
+    private void dealWithReleaseLinksContainedInComponent(Map<String, Component> componentsById,
+            Map<String, Release> releasesById, Component component, List<Iterable<String>> csvRows) {
+        final List<Release> releases = component.getReleases();
+        if (releases != null && !releases.isEmpty()) {
+            for (Release release : releases) {
+                dealWithReleaseLinksContainedInRelease(componentsById, releasesById, component, release, csvRows);
+            }
+        }
+    }
+
+    private void dealWithReleaseLinksContainedInRelease(Map<String, Component> componentsById,
+            Map<String, Release> releasesById, Component component, Release release, List<Iterable<String>> csvRows) {
+        final Map<String, ReleaseRelationship> releaseIdToRelationship = release.getReleaseIdToRelationship();
+        if (releaseIdToRelationship != null) {
+            for (Map.Entry<String, ReleaseRelationship> idReleaseRelationshipEntry : releaseIdToRelationship.entrySet()) {
+                final Release linkedRelease = releasesById.get(idReleaseRelationshipEntry.getKey());
+                if (linkedRelease != null) {
+                    final ReleaseRelationship relationship = idReleaseRelationshipEntry.getValue();
+                    final Component linkedComponent = componentsById.get(linkedRelease.getComponentId());
+                    if (linkedComponent != null) {
+                        printReleaseLinkEntry(component, release, linkedRelease, relationship, linkedComponent, csvRows);
+                    }
+                }
+            }
+        }
+    }
+
+    private void printReleaseLinkEntry(Component component, Release release, Release linkedRelease,
+            ReleaseRelationship relationship, Component linkedComponent, List<Iterable<String>> csvRows) {
+        final ReleaseLinkCSVRecordBuilder releaseLinkCSVRecordBuilder = ReleaseLinkCSVRecord.builder();
+        releaseLinkCSVRecordBuilder.fill(component);
+        releaseLinkCSVRecordBuilder.fill(release);
+        releaseLinkCSVRecordBuilder.fillLinking(linkedRelease);
+        releaseLinkCSVRecordBuilder.fillLinking(linkedComponent);
+        releaseLinkCSVRecordBuilder.setRelationship(relationship);
+        csvRows.add(releaseLinkCSVRecordBuilder.build().getCSVIterable());
+    }
+
+    public void getComponentDetailedExport(User sw360User, HttpServletResponse response)
+            throws TException, IOException {
```

Comment:
Remove the declaration of thrown exception '**TException**', as it cannot be thrown from method's body.